### PR TITLE
Code revision

### PR DESCRIPTION
**Pull request by Bonface-Osoro**
Addressed Point 1, 2 and 3.

### CODE REVIEW

**Comment by Bonface-Osoro** [comment]
Recalculated the monthly traffic

File: scripts/run.py [hunk 1]
```diff
@@ -460,10 +460,10 @@ def process_mission_total():
         'total_climate_change', 'total_climate_change_wc']]
 
     #Create columns to store new data
-    df[["capacity_per_user", "emission_per_capacity", "per_cost_emission", 
-        "per_subscriber_emission", "capex_per_user", "opex_per_user", 
+    df[["capacity_per_user", "per_subscriber_emission", 
+        "capex_per_user", "opex_per_user", "monthly_gb",
         "tco_per_user", "capex_per_capacity", "opex_per_capacity",
```
Comment (edwardoughton) [inline scripts/run.py:465]: Where do we use capex_per_capacity? I think this is wrong. Double check and triple check all metrics, removing any redundant/superfluous metrics.

File: scripts/run.py [hunk 2]
```diff
@@ -525,7 +521,7 @@ def capacity_subscriber(const_cap, subscribers):
     cap_sub : float
         Capacity per subscriber
     """
-    cap_sub = const_cap * 0.5 * subscribers
+    cap_sub = const_cap * 0.5 / subscribers
```
Comment (edwardoughton) [inline scripts/run.py:524]: What is 0.5? This should be in the parameters dictionary with a proper key. Please make sure all parameters are in the parameters dictionary.

File: scripts/run.py [hunk 3]
```diff
@@ -547,7 +543,7 @@ def emission_capacity(clim_emissions, monthly_traffic):
     emission_cap : float
         Emission per capacity
     """
-    emission_cap = (clim_emissions / monthly_traffic) * 12 * 5
+    emission_cap = (clim_emissions / monthly_traffic) / (12 * 5)
```
Comment (edwardoughton) [inline scripts/run.py:546]: What is 5? This should be in the parameters dictionary with a proper key e.g, "time_horizon". Please make sure all parameters are in the parameters dictionary.

**Review by edwardoughton — CHANGES_REQUESTED** [review]
Good work on correcting the metric miscalculation - that looks to be correct again now. 

I've reviewed the rest of the code. Please carry out the following changes:

- Remove superfluous metrics - see comments as examples. Check all metrics. There are more than I identify, so you need to carefully review the whole codebase.
- Lines 341-370 are not PEP8 style and I've asked for this previously to be corrected. Please write in pythonic style as we need to show to reviewers we are details-focused. 
- There are still undefined parameters. All parameters have to be in the parameters dict with a proper key and value. If a function then uses them, you pass the params dict to it to access these. Don't hard code them in, unless it's something as obvious as the number of months in a year, in which case you basically state that in your docstring explanation (and many functions still lack proper docstring explanations imo). 

Please let me know once complete. 

Thanks

**Comment by Bonface-Osoro** [comment]
I have edited the code in PEP8 style

File: scripts/run.py (hunk 1 above)
Comment (edwardoughton) [reply scripts/run.py]: Complete.

File: scripts/run.py (hunk 3 above)
Comment (edwardoughton) [reply scripts/run.py]: I will accept this for now, but we still need to use a parameters dictionary more thoroughly in the codebase, as it helps you and the reviewers have all the key model inputs in one place.

**Pull request merged**